### ISSUE TITLE
Execution Block ID uid://A001/X15a0/X15a Sgr_A_st_af_03_TM1

### ISSUE THREAD

**Issue by keflavich** [ACES-CMZ/reduction_ACES#255]
Sgr_A_st_af_03_TM1
[uid://A001/X15a0/X15a](https://almascience.org/aq/?result_view=observation&mous=uid://A001/X15a0/X15a)

* [x] Observations completed?
* [ ] Delivered?
* [ ] Downloaded? (specify where)
  * [ ] hipergator
  * [ ] hipergator pipeline run

* [ ] [Weblog](https://data.rc.ufl.edu/secure/adamginsburg/ACES/weblogs/humanreadable/Sgr_A_st_af_03_TM1/html/) unpacked
* [ ] [Weblog](https://data.rc.ufl.edu/secure/adamginsburg/ACES/weblogs/humanreadable/Sgr_A_st_af_03_TM1/html/) Quality Assessment?
* [ ] Imaging: Continuum
* [ ] Imaging: Lines

  * [ ] SPW25 H13CN
  * [ ] SPW27 H13CO+/SiO
  * [ ] SPW29 HCO+
  * [ ] SPW31 HNCO
  * [ ] SPW33 Cont 1
  * [ ] SPW35 Cont 2

**Comment by d-l-walker** [comment]
# Calibration
#### Tsys contamination was not flagged out in any SPWs
![tsys-uid___A002_Xfe62c1_X10de7 ms-summary spw19](https://user-images.githubusercontent.com/13015266/213178356-e318aca4-4081-4687-a25b-335e20ed32a2.png)

#### The bulk of the phase absorption of the phase calibrator in SPW 29 was flagged about, but a small, less significant component was missed, e.g.:
![uid___A002_Xfe62c1_X10de7 ms-J1744-3116-spw29-PHASE-amp_vs_freq-XX_YY](https://user-images.githubusercontent.com/13015266/213178370-4947c936-f1c8-42ef-a59e-d1be43ded48d.png)

# Imaging

### Primary beam map looks weird? (Same for all SPWs) 
![uid___A001_X15a0_X15a s38_0 Sgr_A_star_sci spw27 cube I iter1 pb I mean sky](https://user-images.githubusercontent.com/13015266/213139717-5f7ecaa2-14d3-4f61-a683-7768b134d7a3.png)

### SPWs 31 & 35 diverged. SPW 33 was not imaged due to size mitigation.
![uid___A001_X15a0_X15a s38_0 Sgr_A_star_sci spw31 cube I iter1 image pbcor I mom8 sky](https://user-images.githubusercontent.com/13015266/213152125-e81a2c22-e517-4799-b567-3ff85994992e.png)
![uid___A001_X15a0_X15a s38_0 Sgr_A_star_sci spw35 cube I iter1 image pbcor I mom8 sky](https://user-images.githubusercontent.com/13015266/213152138-67ecd257-5510-4ea3-9f5c-f451f8116f5e.png)

**Comment by keflavich** [comment]
@d-l-walker I assigned this to you because you've already done the QA. But we need someone to do the reimaging still.

**Comment by keflavich** [comment]
spw33+35 continuum reduction was done incorrectly; it used the uncorrected data column.  Re-running.

**Comment by keflavich** [comment]
Wild divergence in spw31 (HNCO) at the band edge

<img width="1719" alt="image" src="https://user-images.githubusercontent.com/143715/232914884-b3ad4043-1dab-40f4-b94b-adda84e82e22.png">

**Comment by keflavich** [comment]
I think we concluded today that the sinusoidal features in frequency are not caused by tclean divergence but are present in the UV data.

**Comment by djeff1887** [comment]
Update on spw35: after trying `cyclefactor` 2 and 3, 4 has improved the divergence, but it's still present in a couple channels. Will try `cyclefactor=5` and see how that goes for next week.
![spw35_cf4_stilldiverges](https://github.com/ACES-CMZ/reduction_ACES/assets/62924149/3b2b1b66-3ad3-4f69-80ac-4c1f7f94124e)

**Comment by djeff1887** [comment]
`cyclefactor=5` seems to have done the trick. Pull request for spw35 incoming
![image](https://github.com/ACES-CMZ/reduction_ACES/assets/62924149/ce7f3169-47c0-4892-b9f5-89497c3d6c6a)

**Comment by ashleythomasbarnes** [comment]
Checked weblogs and didn't find anything obvious that would cause this... Indeed as @d-l-walker pointed out, there is some contamination in the Tsys spectra, but I don't think this would have caused the issue. You can also see this in the findcont stage so indeed I think this is a problem and worth sending to ALMA. 

https://github.com/ACES-CMZ/reduction_ACES/issues/255#issuecomment-1513843423 

![Screenshot 2023-08-23 at 16 57 27](https://github.com/ACES-CMZ/reduction_ACES/assets/30494539/2950cdb7-a715-487b-8f44-d2155028a133)

**Comment by d-l-walker** [comment]
Flagging `correlation='YY'` seems to fix the issue.
Note that this is likely throwing away plenty of good data -- all of the bad data were YY, but not all YY points were bad data.
Perhaps we can re-image SPW 31 with this flagged EB for the time-being?

![Screenshot 2023-09-12 at 11 48 19](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/253a845c-0b60-489b-ac0e-c4dbd38986e2)

**Comment by d-l-walker** [comment]
Dirty image of SPW 31 with `correlation='YY'` flagged for EB `uid___A002_Xfe83cd_X286_target.ms` looks much more reasonable, no crazy divergence. 

I'll set a full clean of this running now.

**Comment by xinglunju** [comment]
Are you using stokes='pseudoI' with tclean? By default tclean does not consider the other polarizations if one polarization is flagged... https://casadocs.readthedocs.io/en/stable/api/tt/casatasks.imaging.tclean.html#stokes

**Comment by d-l-walker** [comment]
@xinglunju huh, I wasn't aware of that, thanks!

No, I'm not doing that. But the resulting image looks fine (peak intensity below) ...
<img width="408" alt="Screenshot 2023-09-13 at 06 34 45" src="https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/3415f88f-c9d0-433a-bf37-b6ffdfe975d2">

I can re-run with that option enabled just to be safe.

**Comment by xinglunju** [comment]
@d-l-walker the images look nice!

Is the rms level higher than expected, though? Without stokes='pseudoI', for SPW31 we are using 4/6 of the data (or 2 out of the 3 exection blocks), and with pseudoI, we will use 5/6, so the rms may be different by ~10%.

**Comment by d-l-walker** [comment]
Yeah I agree that's likely what's happening based on the warning you linked. I set the clean re-running earlier, so I'll check when it's finished in a few hours and compare it with the image without pseudoI.

**Comment by snlongmore** [comment]
The following has been submitted as a Helpdesk ticket [22864](https://help.almascience.org/tickets/22864):

We have found the following problems with the 12m data for the ACES field "af":

- Spectral window 31, which contains HNCO (our primary kinematic tracer), contains bad data that significantly affect the images.
- We found very strong divergence in the field, and after some digging we realised that this was not an imaging problem, but a data problem.
- The problem is specifically with one execution block, uid___A002_Xfe83cd_X286.
- Towards the upper edge of the frequency band, the amplitude of the data rises dramatically.
- Flagging out all data corresponding to correlation = 'YY' successfully removes the bad data and the resulting image cube looks good with no artifacts.
- All of the bad data points appear to be correlation = 'YY', but not all YY data are bad, so more careful flagging is needed to avoid throwing out good data.
- Beyond that we're not sure what the cause of this is. Presumably hardware related?
- A more detailed discussion of the problem, along with various plots and figures can be found in our Github issue for this field: https://github.com/ACES-CMZ/reduction_ACES/issues/255

We would appreciate your help diagnosing (and ideally solving) the problem.

**Comment by ashleythomasbarnes** [comment]
**Helpdesk ticket response:** 

The QA3 for this MOUS is approved: https://jira.alma.cl/browse/PRTSPR-70155
However, take into account that the EB providing the not-uniform PB is due to the EB uid___A002_Xfe62c1_X10de7 that was partially observed.
You are asking to check the data of uid___A002_Xfe83cd_X286, that at the moment does not seem to me an issue. I will update you further as soon as possible.

**Comment by keflavich** [comment]
while spw31 is in QA3, spw33 also had divergence.  https://github.com/ACES-CMZ/reduction_ACES/pull/408/commits/c76e57abb8bb04d7813caa992841ddfec1ac27ff proposes a change to fix that.

**Comment by djeff1887** [comment]
Cont QA (spw22_25,spw33_35): I'm assuming most of this is free-free filaments, but there does appear to be some slight line contamination in addition to the filaments in both images here too.
![image](https://github.com/ACES-CMZ/reduction_ACES/assets/62924149/bfc34fb6-c6d9-43fb-82dd-bfa836caf9be)

**Comment by d-l-walker** [comment]
I'm just making a note that `uid___A001_X15a0_X15a.s38_0.Sgr_A_star_sci.spw35.cube.I.iter1.image.pbcor.statcont.contsub.fits` still has divergence. @djeff1887 fixed this and added `cyclefactor=5` to the override commands, but the existing version used 1.5 (according to header history).

Typically we would have to re-run and re-statcont SPW 35 to fix this, however, we still need to download the post-QA3 data and re-run the PL (@keflavich), so no action needed yet. Note that SPW 35 still diverged in the QA3 PL run.

**Comment by keflavich** [comment]
If @djeff1887 has a cleaned version that's good to go on disk, and it's unaffected by QA3 (wasn't that for another window?), we can just move that over - @djeff1887 where is it?

**Comment by djeff1887** [comment]
@keflavich it's here: `/orange/adamginsburg/sgrb2/d.jeff/X15a/calibrated/working/spw35cycle5/`

**Comment by keflavich** [comment]
Trying to run the new pipeline, got this failure:
```
CASA 6.4.1.12 -- Common Astronomy Software Applications [6.4.1.12]
*** ALMA scriptForPI ***
Found more than one piperestorescript:
['member.uid___A001_X15a0_X15a.hifa_calimage_selfcal.casa_piperestorescript.py', 'member.uid___A001_X15a0_X15a.hifa_calimage.casa_piperestorescript.py']
ERROR: non-unique piperestorescript
```
@d-l-walker this looks like a bug?

**Comment by d-l-walker** [comment]
@keflavich the data were re-reduced using a newer version of the PL for QA3 (v6.5.4.9), so I think you'll need to re-run with this version. 

It looks like there's a script relating to selfcal. This is present in the new PL, but not supported for mosaics and so wasn't performed here, so I'm not 100% sure what the difference would be between these two. I'll look into it.

**Comment by keflavich** [comment]
@d-l-walker is the required CASA version stored anywhere in machine-readable format?

**Comment by d-l-walker** [comment]
@keflavich it should be in the QA2 report, but that's a .pdf file, so not the most easily machine-readable. Probably somewhere in the weblog files, I think you can find it in the base index.html file. I'm not sure if it's given anywhere else.

**Comment by keflavich** [comment]
the CASA version isn't the issue, the problem is still the same:
```
CASA <3>: %run -i member.uid___A001_X15a0_X15a.scriptForPI.py
*** ALMA scriptForPI ***
Found more than one piperestorescript:
['member.uid___A001_X15a0_X15a.hifa_calimage_selfcal.casa_piperestorescript.py', 'member.uid___A001_X15a0_X15a.hifa_calimage.casa_piperestorescript.py']
An exception has occurred, use %tb to see the full traceback.

SystemExit: ERROR: non-unique piperestorescript
```

Should we be using the selfcal version or the non-selfcal version?  And, why were the data distributed with invalid file names?

**Comment by keflavich** [comment]
the QA2 PDF does not mention selfcal

**Comment by keflavich** [comment]
OK, I think that indicates some kind of error in the data distribution.

I moved all the `selfcal` files to a subdirectory of the script directory called `bad/` and triggered the scriptForPI.

**Comment by keflavich** [comment]
Pipeline run finished with:
```
2024-04-30 09:14:53 INFO: Selecting representative target source Sgr_A_star for data set uid___A002_Xfe62c1_X10de7.ms                                                                                                                                                                         |
2024-04-30 09:14:53 INFO: Selecting representative target source Sgr_A_star for data set uid___A002_Xfe83cd_X286.ms                                                                                                                                                                           |
2024-04-30 09:14:53 INFO: Selecting representative target source Sgr_A_star for data set uid___A002_Xfe83cd_X125c.ms                                                                                                                                                                          |
2024-04-30 09:14:54 INFO: Saving context: pipeline-20240429T155240.context                                                                                                                                                                                                                    |
Imaging pipeline was used. Will not create uid___A002_Xfe62c1_X10de7.ms.split.cal                                                                                                                                                                                                             |
Linking MS uid___A002_Xfe62c1_X10de7.ms into directory "calibrated"                                                                                                                                                                                                                           |
Imaging pipeline was used. Will not create uid___A002_Xfe83cd_X286.ms.split.cal                                                                                                                                                                                                               |
Linking MS uid___A002_Xfe83cd_X286.ms into directory "calibrated"                                                                                                                                                                                                                             |
Imaging pipeline was used. Will not create uid___A002_Xfe83cd_X125c.ms.split.cal                                                                                                                                                                                                              |
Linking MS uid___A002_Xfe83cd_X125c.ms into directory "calibrated"                                                                                                                                                                                                                            |
Done. Please find results in directory "calibrated".              
```
Should be OK, maybe?

**Comment by keflavich** [comment]
Continuum reclean.  25+27 in the top left, 33+35 in top right, all windows in bottom left
<img width="927" alt="image" src="https://github.com/ACES-CMZ/reduction_ACES/assets/143715/5f8d64df-b228-4c2d-af19-4cf76efa45db">
<img width="933" alt="image" src="https://github.com/ACES-CMZ/reduction_ACES/assets/143715/558102c7-a8b5-4b14-b5a6-66f62a805b5f">

I've highlighted some sources that show up in 25+27 but not 33+35.  The noise appears higher in 33+35.  Both of these suggest to me that there's a problem with the continuum identification.

**Comment by d-l-walker** [comment]
@keflavich I'm just checking on the status of the post-QA3 cleaning, and the data still look like pre-QA3? SPW 31 still has the crazy divergence/ringing. The file that I grabbed was `~/member.uid___A001_X15a0_X15a/calibrated/working/uid___A001_X15a0_X15a.s38_0.Sgr_A_star_sci.spw31.cube.I.iter1.image.pbcor.statcont.contsub.fits`, which was created 2 days ago.

Not sure what's going on here. I feel like you downloaded this previously and it was still the pre-QA3 data ... 🤔

<img width="1342" alt="Screenshot 2024-05-09 at 11 19 21" src="https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/e4df436f-a441-43d2-880b-6e1012aedf43">

**Comment by keflavich** [comment]
Yep, this is still the pre-QA3 data.  Maybe I was supposed to keep the selfcal version?

At this point, I think we need to file a ticket, but it would be helpful if we could get someone else to try to restore these data locally and confirm my confusion.

**Comment by d-l-walker** [comment]
I'd be happy to try, but I need access to the data. Can you do that? Or does Steve need to add me as a delegee?

**Comment by ashleythomasbarnes** [comment]
Any update on fixing the QA3 issue with the reduction rerun @d-l-walker ?

**Comment by d-l-walker** [comment]
@ashleythomasbarnes I can confirm that everything looks good

Left = post-QA3 image, Right = old image
(note: it looks works because I degraded the resolution to speed it up)

![Screenshot 2024-07-24 at 13 43 26](https://github.com/user-attachments/assets/f7b7242a-30d5-4d83-9f41-b25c764885c3)

@keflavich not sure what happened on your end, but I'm guessing some issue with the bookkeeping of old and new data. Do you want to try again? Or do you want me to re-do this on my end? (I'll have to re-run the imaging steps again to image at full res, and to account for our internal changes such as `fitorder=0`)

**Comment by ashleythomasbarnes** [comment]
Thanks @d-l-walker, this looks good! Are we then okay to close the ticket?

**Comment by keflavich** [comment]
Mosaicking the HCNO chunk of this is still failing, even in micro tests

<img width="1022" alt="image" src="https://github.com/user-attachments/assets/d9b51d86-7bad-4de7-bcee-d2c4a080f722">
<img width="879" alt="image" src="https://github.com/user-attachments/assets/18bc9230-6b8f-4dce-953c-23d83cdbe786">

I have no explanation yet.

**Comment by d-l-walker** [comment]
@keflavich did you ever manage to resolve the issue with the QA3 data for SPW 31 (HNCO)? Previously it was still looking like the broken pre-QA3 data despite using the new data.

As noted above, the cube in the delivered product folder looks good. These are the MSs that should be used for imaging the cube:

`['uid___A002_Xfe62c1_X10de7_targets_line.ms', 'uid___A002_Xfe83cd_X286_targets_line.ms', 'uid___A002_Xfe83cd_X125c_targets_line.ms']`

This MS naming convention will be different to the pre-QA3 data due to the use of the newer PL during QA3.

**Comment by d-l-walker** [comment]
The affected vel/freq range for the diverging channels in SPW 35 is: 
`100.0555 GHz; Velocity: 1325.8439 km/s` to `100.0629 GHz; Velocity: 1303.9943 km/s `
